todo/src/reducers/index.js

```diff
@@ -0,0 +1,34 @@
+import {ADD_TODO, TOGGLE_COMPLETE, DELETE_TODO} from '../actions/index';
+
+
+const initialState = {
+    todos: []
+}
+
+export const toDoReducer = (state = initialState, action) => {
+    switch(action.type){
+        case ADD_TODO:
+        return state = { todos: [...state.todos, action.payload]};
+        case TOGGLE_COMPLETE: 
+        let todos = state.todos.map(todo =>{ 
+            if(todo.id === action.payload){
```

Comment:
you should be copying the todo instead of mutating it. Like this:

return { ...todo, completed: !todo.completed }